### ISSUE TITLE
Feature Request: Support the use of jetbrains "external annotations" annotation.xml format for stubbing.

### ISSUE THREAD

**Issue by sir-maniac** [typetools/checker-framework#6258]
ItelliJ Idea keeps track of external annotations using a simple xml format and directory tree.  

The documentation can be found [here](https://www.jetbrains.com/help/idea/annotating-source-code.html#external-annotations) .

Examples of the format can be found [here](https://github.com/JetBrains/intellij-community/tree/39fa92a0ad83a4ad060d0e37e2a09c7fc180c49d/lib/annotations) in the [intellij community source code](https://www.jetbrains.com/help/idea/annotating-source-code.html#external-annotations)

For example, to stub java.lang.String:
File: `java/lang/annotations.xml`
```xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
  <item name="java.lang.String java.lang.String intern()">
    <annotation name="org.checkerframework.checker.interning.qual.Interned" />
  </item>
</root>
```
The xml format is simple so it should be easy to implement.

This would make it easier to work with this library for IntelliJ users, as they can easily add the information through the IDE.  It is also easy to share with others, and can be distributed like a source jar.  

This is likely more performant than stubs as well.

**Comment by mernst** [comment]
That's a cool idea.  We would gladly accept a pull request.  I wish that the format was actually documented...

**Comment by sir-maniac** [comment]
Probably the best 'document' would be the unit tests.  A [cursory search](https://github.com/search?q=repo%3AJetBrains%2Fintellij-community+path%3A%2F%5Ejava%5C%2Fjava-tests%5C%2FtestSrc%5C%2Fcom%5C%2F%2F+externalannotations&type=code) shows just one main [test class](https://github.com/JetBrains/intellij-community/blob/71ebe8cf62367e10651766c18a87cf2b66b78351/java/java-tests/testSrc/com/intellij/java/codeInsight/ExternalAnnotationsTest.java).  

Most of the action seems to be handled in [ExternalAnnotationsManager](https://github.com/JetBrains/intellij-community/blob/71ebe8cf62367e10651766c18a87cf2b66b78351/java/java-psi-api/src/com/intellij/codeInsight/ExternalAnnotationsManager.java).

I won't promise, but I'll spend a little time and see what I can do myself.

**Comment by 555vedant** [comment]
Please can you asssign this issue to me i wanted to work with this issue

**Comment by ceasermikes002** [comment]
Hi @sir-maniac  and @mernst , I'd like to work on this issue. I'll take a look at the existing stub-file implementation and implement support for the IntelliJ external annotations XML format.

**Comment by ceasermikes002** [comment]
Hi @jayvdb  and @sir-maniac, I have completed the implementation and pushed the commits. I would greatly appreciate it if you could review my work when you have a chance. Please let me know if you need any changes!
